src/material-experimental/mdc-form-field/form-field.ts

```diff
@@ -192,11 +192,11 @@ export class MatFormField implements AfterViewInit, OnDestroy, AfterContentCheck
   }
   private _hintLabel = '';
 
-  // Unique id for the hint label.
-  _hintLabelId: string = `mat-hint-${nextUniqueId++}`;
+  /** Unique id for the label content. */
+  readonly labelContentId: string = `mat-form-field-label-${nextUniqueId++}`;
```

Comment:
I'm not sure how intuitive this would be as a public API, because:
1. The element that it's pointing might not always exist since it depends on `_hasFloatingLabel`, which would make accessibility worse.
2. It assumes that people are familiar with the internal markup of the form field which we could change.

Reply:
I agree that (1) is a good point. Regardless of that though, we should decide what to do here. I don't think it's _correct_ to expose the label id privately if it is used by controls such as the select. Obviously there is a need for this.

If no explicit `mat-label` is specified, it's up to the form control to manage accessibility in terms of `aria-label`/`aria-labelledby`. We can never guarantee that a label is provided from within the form-field, but we can provide them with 
a reference to the label element if it's available. I agree that it can be confusing though if no label is present. We could make that obvious by returning `null` if so.

For (2): I don't follow why people would need to be familiar with the internal markup of the form-field? We just provide them with the id to the label element if they need it. This is in particular necessary for custom form-field controls that need special accessibility treatment. That's why this id seems to actually exist. It has been added for the select. I think it must be obvious for authors of custom form-field controls that there is a label (if they care about accessibility).

Would love to chat more about this in the team meeting, or here on the issue. Do you have any proposals?

Reply:
It's just my two cents, I don't feel strongly against it. We can see what the others think.

Reply:
Put this on the agenda for the team meeting?

Reply:
added.

Reply:
as discussed in the meeting, lets call it `labelId`